### PR TITLE
Add field definition for empty space

### PR DESCRIPTION
**Pull request by marckhouzam**
Q | A
-- | --
Bug fix? | no
New feature? | yes
BC breaks? | no
Deprecations? | no
Tests pass? | yes
Fixed tickets | #217

Add a new field type for inserting empty space in custom object forms. For details, see ticket #217

### CODE REVIEW

**Comment by marckhouzam** [comment]
@loicbenoit is the developer behind this PR

**Comment by marckhouzam** [comment]
Sorry. We'll have a look.

**Comment by marckhouzam** [comment]
Any comments on this updated PR?  Thanks!

**Comment by loicbenoit** [comment]
Is it relevant?
The goal is to improve the user experience in the following ways:
1. When multiple independant objects have some fields in common, it is less confusing to always display these common fields in the expected column and relative order.
2. This PR provides a very simple way to let users control the form layout without rewriting the entire plugin, nor reinventing the way GLPI displays forms. Non-technical users can control the layout using existing GUI tools, already provided by the plugin and fully compatible with existing features.

Is it correct?
As explained in ticket #217, this is not ideal, but inline with GLPI architecture. GLPI is deeply relationnal and not MVC. Thus, the form layout directly reflects the table structure, up to the relative order of fields in the DB table. Adding a table column for empty fields is thus a natural solution that lets the relationnal model directly reflect the state of the custom object.

Hoping this helps.

**Comment by marckhouzam** [comment]
Any thoughts after the provided explanations?
Thanks!

**Comment by loicbenoit** [comment]
Following up to see if anyone can approve this request or if you have more questions.

File: inc/object.class.php [hunk 1]
```diff
@@ -522,6 +522,11 @@ function displayField($canedit, $name, $value, $template, $description = []) {
       if (!empty($searchoption)
          && !in_array($name, self::getFieldsToHide())) {
 
+         if (strpos($description['Field'], 'plugin_genericobject_emptyspace') === 0) {
```
Comment (cedric-anne) [inline inc/object.class.php]: I suggest replacing the test on name by following test. To be able to use any name for spacers.

```suggestion
         if (isset($searchoption['input_type']) && 'emptyspace' === $searchoption['input_type']) {
```

**Comment by cedric-anne** [comment]
Hi,

I'm going to merge this PR for next release of this plugin.
Please open a PR for documentation update (file: https://github.com/pluginsGLPI/doc/blob/master/source/genericobject/index.rst).

Regards

**Pull request merged**